### ISSUE TITLE
In `expr.str.slice()` indicate whether an index of 0 or 1 means "start at the start of the string"

### ISSUE THREAD

**Issue by DeflateAwning** [pola-rs/polars#16413]
### Description

I always have trouble figuring out how to use slice as I mentally switch between Rust, Python, Ruby, C, and others, each with their own indexing systems.

Perhaps adding a column example which slices `.str.slice(1, 3)` to the example with "payaya" would be helpful. 

Idea: name the columns literally `".str.slice(1, 3)"`, `".str.slice(4, 3)"` for clarity. 

### Link

https://docs.pola.rs/py-polars/html/reference/expressions/api/polars.Expr.str.slice.html#polars.Expr.str.slice

**Comment by orlp** [comment]
> as I mentally switch between Rust, Python, Ruby, C, and others, each with their own indexing systems

All of these use zero-based indexing. I don't think this needs explaining, especially because the host language, Python, uses zero-based indexing.

**Comment by DeflateAwning** [comment]
Yeah fair enough - have gotten bitten by SQL's `MID()` function enough times that I trust nothing.
